week-1-ts-ecosystem/w01e03-object-oriented-structure/src/basket/basket.ts

```diff
@@ -0,0 +1,50 @@
+import {Product} from "../product/product";
+import {ProductId} from "../product/models/product-properties.model";
+
+export class Basket<Type extends Product> {
+    private products: Map<ProductId, Type> = new Map<ProductId, Type>();
+
+    addProduct(product: Type) {
+        this.products.set(product.id, product);
+    }
+
+    getProductById(id: ProductId) {
+        return this.products.get(id);
+    }
+
+    increaseAmount(id: ProductId, amount: number) {
+        this.products.get(id)?.increaseAmount(amount);
+    }
+
+    decreaseAmount(id: ProductId, amount: number) {
+        const product = this.products.get(id);
+        if(product){
+            product.decreaseAmount(amount);
+            if(product.amount <= 0) {
+                this.products.delete(id);
+            }
+        }
+    }
+
+    removeProduct(id: ProductId) {
+        this.products.delete(id);
+    }
+
+    get getAllProducts() {
```

Comment:
nazwałbym to po prostu `allProducts` bez `get` to `get` z przodu sugeruje że to funkcja a nie pole